src/tasks/JsonToItemsTaskFactory/JsonToItemsTaskFactory.cs

```diff
@@ -353,20 +370,20 @@ public override JsonModelItem Read(ref Utf8JsonReader reader, Type typeToConvert
                     case JsonTokenType.String:
                         var stringItem = reader.GetString();
                         if (string.IsNullOrEmpty(stringItem))
-                            throw new Exception ("deserialized json string item was null or the empty string");
+                            throw new JsonException ("deserialized json string item was null or the empty string");
                         return new JsonModelItem(stringItem!, metadata: null);
                     case JsonTokenType.StartObject:
                         var dict = JsonSerializer.Deserialize<Dictionary<string, string>>(ref reader, options);
                         if (dict == null)
                             return null!;
                         var idict = new Dictionary<string, string> (dict, StringComparer.OrdinalIgnoreCase);
                         if  (!idict.TryGetValue("Identity", out var identity) || string.IsNullOrEmpty(identity))
-                            throw new Exception ("deserialized json dictionary item did not have a non-empty Identity metadata");
+                            throw new JsonException ("deserialized json dictionary item did not have a non-empty Identity metadata");
                         else
                             idict.Remove("Identity");
                         return new JsonModelItem(identity, metadata: idict);
                     default:
-                        throw new Exception();
+                        throw new NotSupportedException();
```

Comment:
token type in the message might be useful for debugging

Reply:
The JsonSerializer catches and decorates `NotSupportedException` with location and type info https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-converters-how-to?pivots=dotnet-5-0#notsupportedexception